src/persist/EpochAdvancers.cpp

```diff
@@ -89,8 +90,14 @@ void DedicatedEpochAdvancer::advancer(int task_num){
 }
 
 void DedicatedEpochAdvancer::sync(uint64_t c){
-    uint64_t target_epoch = (c == NULL_EPOCH)? esys->get_epoch()+2 : c+2;
+    // sync must NOT be called in an operation.
```

Comment:
I thought `sync` takes target epoch number (or target-2)? Why passing it if it's ought to be null?

Reply:
`Recoverable::sync()` takes no argument; `EpochcSys::sync()` will only be called internally. Passing the argument in helps with verifying, since calling sync in operations will cause deadlock.